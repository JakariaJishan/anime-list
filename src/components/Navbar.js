import React from 'react'
import { NavLink } from 'react-router-dom'

function Navbar() {
  return (
    <div>
      <NavLink to='/' >Home</NavLink>
    </div>
  )
}

export default Navbar