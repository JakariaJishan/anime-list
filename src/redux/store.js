import { configureStore } from '@reduxjs/toolkit';
import animeSlice from './Anime-list/animeSlice';
import episodeSlice from './Anime-list/episodeSlice';

export default configureStore({
  reducer: {
    animeList: animeSlice,
    episodeList: episodeSlice,
  },
});
