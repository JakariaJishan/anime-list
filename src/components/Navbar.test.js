import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Navbar from './Navbar';

describe('Navbar test case', () => {
  test('Navbar test the snapshot case', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
