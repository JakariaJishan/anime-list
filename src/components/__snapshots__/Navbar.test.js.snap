// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test case Navbar test the snapshot case 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <nav
          class="nav"
        >
          <img
            alt="home icon"
            height="20px"
            src="home.png"
            width="20px"
          />
          <a
            aria-current="page"
            class="brand active"
            href="/"
          >
            ANIME
          </a>
          <div
            class="desktopIcon"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <img
                alt="mic icon"
                height="20px"
                src="microphone-black-shape.png"
                width="20px"
              />
            </a>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <img
                alt="gear icon"
                height="20px"
                src="gear.png"
                width="20px"
              />
            </a>
          </div>
          <a
            aria-current="page"
            class="menu active"
            href="/"
          >
            <img
              alt="menu icon"
              height="20px"
              src="menu.png"
              width="20px"
            />
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <nav
        class="nav"
      >
        <img
          alt="home icon"
          height="20px"
          src="home.png"
          width="20px"
        />
        <a
          aria-current="page"
          class="brand active"
          href="/"
        >
          ANIME
        </a>
        <div
          class="desktopIcon"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <img
              alt="mic icon"
              height="20px"
              src="microphone-black-shape.png"
              width="20px"
            />
          </a>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <img
              alt="gear icon"
              height="20px"
              src="gear.png"
              width="20px"
            />
          </a>
        </div>
        <a
          aria-current="page"
          class="menu active"
          href="/"
        >
          <img
            alt="menu icon"
            height="20px"
            src="menu.png"
            width="20px"
          />
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
